File: problemset/14. Longest Common Prefix/main.py
class Solution:
    def longestCommonPrefix(self, strs: list[str]) -> str:
        ans = ""
        for l in range(0, len(strs[0])):
            for s in strs:
                if not s.startswith(strs[0][:l + 1]):
                    return strs[0][:l]
            ans += strs[0][l]
        return ans

File: problemset/14. Longest Common Prefix/main.js
/**
 * @param {string[]} strs
 * @return {string}
 */
var longestCommonPrefix = function (strs) {
    strs.sort()
    const first = strs[0]
    const last = strs[strs.length - 1]
    let ans = ""
    for (let i = 0; i < Math.min(first.length, last.length); i++) {
        if (first[i] === last[i]) {
            ans += first[i]
        } else {
            break
        }
    }
    return ans
};

console.log(longestCommonPrefix(["dog", "racecar", "car"]))
